script/publish_public_data_json

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python3
+
+"""
+Export query result data as JSON to a publicly accessible bucket.
+
+Data of the query is exported if "json_export" is set in
+the corresponding metadata file.
+"""
+
+from argparse import ArgumentParser
+import os
+import sys
+import re
+
+from google.cloud import bigquery
+
+# sys.path needs to be modified to enable package imports from parent
+# and sibling directories. Also see:
+# https://stackoverflow.com/questions/6323860/sibling-package-imports/23542795#23542795
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+from bigquery_etl.parse_metadata import Metadata  # noqa E402
+
+
+DEFAULT_BUCKET = "ascholtz-dev-test"  # todo
+METADATA_FILE = "metadata.yaml"
+SUBMISSION_DATE_RE = re.compile(r"^submission_date:DATE:(\d\d\d\d-\d\d-\d\d)$")
+QUERY_FILE_RE = re.compile(r"^.*/([a-zA-Z0-9_]+)/([a-zA-Z0-9_]+)_(v[0-9]+)/query\.sql$")
+
+parser = ArgumentParser(description=__doc__)
+parser.add_argument(
+    "--target-bucket",
+    "--target_bucket",
+    default=DEFAULT_BUCKET,
+    help="GCP bucket JSON data is exported to.",
+)
+parser.add_argument(
+    "--project_id",
+    default="moz-fx-data-derived-datasets",
+    help="Run query in the target project",
+)
+parser.add_argument("--parameter", help="Query parameters, such as submission_date")
+parser.add_argument("--query_file", help="File path to query to be executed")
+
+
+def publish_table_as_json(
+    bucket, client, dataset, table, version, result_table, date=None
+):
+    """Export the `result_table` data as JSON to Cloud Storage."""
+    # "*" makes sure that files larger than 1GB get split up into multiple JSON files
+    destination_uri = f"gs://{bucket}/{dataset}/{table}/{version}/files/"
+
+    if date is None:
+        # if date is not set, then export the entire table and overwrite existing JSON
+        destination_uri += "*.json"
+    else:
+        # if date exists, then query is incremental and newest results are exported
+        destination_uri += f"{date}/*.json"
+
+    table_ref = client.get_table(result_table)
+
+    job_config = bigquery.ExtractJobConfig()
+    job_config.destination_format = "NEWLINE_DELIMITED_JSON"
```

Comment:
The spec specifies we use JSON, not ndjson. The reason for this is it's easier for non-engineer types to use, and is easily viewable in the browser. Unfortunately that means we'll need to pull these files down and transform them.